applications/FluidDynamicsApplication/custom_processes/weighted_divergence_calculation_process.h

```diff
@@ -0,0 +1,338 @@
+
+        if (time_step_current >= time_coefficient * final_time) {
+
+            // Check and set number of elements
+            int number_elements;
+            KRATOS_ERROR_IF(mrModelPart.Elements().size() == 0) << "the number of elements in the domain is zero. weighted divergence calculation cannot be applied"<< std::endl;
+            number_elements = mrModelPart.Elements().size();
+
+            // Check and set number of nodes
+            int number_nodes;
+            KRATOS_ERROR_IF(mrModelPart.Nodes().size() == 0) << "the number of nodes in the domain is zero. weighted divergence calculation cannot be applied" << std::endl;
+            number_nodes = mrModelPart.Nodes().size();
+
+            // Current domain size
+            const std::size_t dimension = mrModelPart.GetProcessInfo()[DOMAIN_SIZE];
+
+            // Auxiliar containers
+            Matrix DN_DX, J0;
+            Vector N;
+
+            // Iterate over the elements
+            #pragma omp parallel for firstprivate(DN_DX,  N, J0)
+            for(int i_elem=0; i_elem < number_elements; ++i_elem) {
+                auto it_elem = mrModelPart.ElementsBegin() + i_elem;
+                auto& r_geometry = it_elem->GetGeometry();
+
+                // Current geometry information
+                const std::size_t local_space_dimension = r_geometry.LocalSpaceDimension();
+                // KRATOS_WATCH(local_space_dimension);
+                const std::size_t number_nodes_element = r_geometry.PointsNumber();
+                // KRATOS_WATCH(number_nodes_element);
+
+                // Resize if needed
+                if (DN_DX.size1() != number_nodes_element || DN_DX.size2() != dimension)
+                    DN_DX.resize(number_nodes_element, dimension);
+                if (N.size() != number_nodes_element)
+                    N.resize(number_nodes_element);
+                if (J0.size1() != dimension || J0.size2() != local_space_dimension)
+                    J0.resize(dimension, local_space_dimension);
+
+                // Build values vectors of the velocity
+                Vector values_x(number_nodes_element);
+                Vector values_y(number_nodes_element);
+                Vector values_z(number_nodes_element);
+                for(int i_node=0; i_node < number_nodes_element; ++i_node){
+                    values_x[i_node] = r_geometry[i_node].FastGetSolutionStepValue(VELOCITY_X);
+                    values_y[i_node] = r_geometry[i_node].FastGetSolutionStepValue(VELOCITY_Y);
+                    values_z[i_node] = r_geometry[i_node].FastGetSolutionStepValue(VELOCITY_Z);
+                }
+
+                // Set integration points
+                const auto& r_integration_method = r_geometry.GetDefaultIntegrationMethod(); // Default is 0
+                const auto& r_integration_points = r_geometry.IntegrationPoints(r_integration_method); // Default is [3 dimensional integration point(0.333333 , 0.333333 , 0), weight = 0.5]
```

Comment:
These comments hold for the 3 nodes triangle, but I think that this process supports any geometry. I'd would remove them since they are misleading.

Reply:
Ok